### DISCUSSION TITLE
🌟💶 YOLOv5 🚀 EXPORT Competition!! $10000 in Prizes

### DISCUSSION THREAD

**Discussion by glenn-jocher** [ultralytics/yolov5#3213] [Show and tell]
<img width="1000" alt="YOLOv5 EXPORT Competition" src="https://github.com/ultralytics/yolov5/releases/download/v1.0/splash-export-competition.png">

We are super excited to announce the first-ever **Ultralytics YOLOv5 🚀 EXPORT Competition** with **$10,000.00** in cash prizes!!!

Our goal is to help everyone easily **train** the world's best Vision AI models, and also to help everyone **deploy** their models just as easily everywhere they want to use them. To help us offer the best deployment solutions for YOLOv5 users we decided to open this EXPORT Competition and let our amazing community contribute their own solutions, with prizes for the best submissions!

Based on community feedback, we created **5 categories** which represent the most popular **real-world deployment** scenarios for YOLOv5 models, including Jetson Nano, Raspberry Pi, Google Edge TPU, desktop CPU and Android edge devices.

Competition participants will create **export tutorials** for the category of their choice, including **open-source code** and **full documentation**. Submissions will be judged by the **simplicity** and **reproducibility** of their export methods, the **quality** of their documentation, and the **speed** and **accuracy** of their exported models. 50% of submission scores will be decided by Ultralytics, and 50% will be decided by community feedback (by summing 👍 or 👎)!

First place winners in each category will receive the full prize money of $2000.00. The competition will run from May 17th, 2021 to August 31st, 2021. Full details are below, and any questions can be asked directly in this discussion thread or by starting a new [discussion](https://github.com/ultralytics/yolov5/discussions). Thanks for reading, and good luck 🍀!

## 🤔 Competition Categories

| | Category        | Evaluation hardware         | Prize |
| ------------- |:-------------:|:-------------:|:-------------:|  
| 1 |   [Nvidia Jetson Nano](https://github.com/ultralytics/yolov5/discussions/3426)      | [Jetson Nano Developer Kit](https://developer.nvidia.com/embedded/jetson-nano-developer-kit) | $ 2,000 |
| 2 |   [Google Edge TPU](https://github.com/ultralytics/yolov5/discussions/3428)     | [Coral Dev Board Mini](https://coral.ai/products/dev-board-mini)  | $ 2,000 |
| 3 |   [Raspberry Pi](https://github.com/ultralytics/yolov5/discussions/3430)  | [Raspberry Pi 4 Model B](https://www.raspberrypi.org/products/raspberry-pi-4-model-b/)     |  $ 2,000 |
| 4 |   [Intel/AMD CPU](https://github.com/ultralytics/yolov5/discussions/3432)  | [AWS EC2 t3.medium](https://aws.amazon.com/ec2/instance-types/)   | $ 2,000 |
| 5 |    [Android](https://github.com/ultralytics/yolov5/discussions/3434)  | [Xiaomi Mi 11](https://www.mi.com/global/mi-11/)    | $ 2,000 |

Evaluation hardware will be running the most up-to-date operating systems at time of evaluation. One winner per category. Competitors can participate in more than one category. 

## 💡 Scoring 

50% of submission scores will be decided by Ultralytics, and 50% will be decided by community feedback, by summing 👍 or 👎 on each submission.

Ultralytics scoring will be determined by:
* **Simplicity of export (20%).** The simplest export will have the least number of steps, require the least numbers of arguments/parameters, use the least number of imported packages, and be executable with the smallest amount of code.
* **Quality of documentation (20%).** Submissions should be well documented using either a **markdown tutorial** (i.e. https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data) or **docs** (i.e. https://docs.ultralytics.com). Each step should be explained, including setup/requirements, any settings/arguments, export steps, and deployed environment setup, if applicable. 
* **Completeness of submission (20%).** Every aspect of export and deployment, starting from an official `yolov5s.pt` model, should be included. For environments that require special requirements, like Jetson Nano, all packages and/or Docker images must be supplied and documented. For Android deployments, an Android reference app should also be included. A submission must include 100% of what is required to **fully export and use** a YOLOv5 model.
* **Speed and accuracy of the deployed model (40%).** Deployed models should return near-identical inference results to official YOLOv5 PyTorch models (i.e. inference with `python detect.py --weights yolov5s.pt`). **Accuracy** of deployed solutions will be analyzed on a hold-out test set of Ultralytics images which are unavailable to the public. **Speed** is very important also, with **the fastest deployment solutions heavily favored**. For Android, exports to [GPU](https://www.tensorflow.org/lite/performance/gpu), [NNAPI](https://www.tensorflow.org/lite/performance/nnapi) and [Hexagon](https://www.tensorflow.org/lite/performance/hexagon_delegate) delegates will receive highest scoring here.

## 💶 Prizes $10000.00

The best submission in each of the 5 Categories will claim the full prize funds of **$2000.00 (2000.00 USD)** from Ultralytics for that category. Funds will be converted to the participant's local currency using the current exchange rate on the date of transfer. Prize money will be transferred via [Wise](https://wise.com/), see the country eligibility [list](https://wise.com/help/articles/2571942/what-countries-can-i-send-to) for prize money transfers.

## 📅 Deadline

The deadline for submissions is 24:00 UTC **August 31st, 2021**. After this date, the competition will be closed and further submissions will not be eligible for prize money. If multiple submissions are scored identically then the submission with the earliest timestamp will act as a tie-breaker.

## 😃 Submissions

To participate, create a public Github repository for your submission, assign your work an [open source license](https://choosealicense.com/) and post your submission directly to one of the 5 official EXPORT Competition submissions threads to allow the community to vote. Note these threads are only for official submissions. General questions or comments may be asked directly in this thread, or in a new [discussion](https://github.com/ultralytics/yolov5/discussions). 

Links to submissions:
| | Category        |
| ----- | ------------- |
| 1 |  [Nvidia Jetson Nano](https://github.com/ultralytics/yolov5/discussions/3426)   |
| 2 | [Google Edge TPU](https://github.com/ultralytics/yolov5/discussions/3428)    |
| 3 | [Raspberry Pi](https://github.com/ultralytics/yolov5/discussions/3430)    |
| 4 | [Intel/AMD CPU](https://github.com/ultralytics/yolov5/discussions/3432)    |
| 5 | [Android](https://github.com/ultralytics/yolov5/discussions/3434)    |

Evaluation will take place from September 1, 2021, to September 16, 2021. Winners will be announced in late September 2021, with prizes disbursed immediately thereafter. 

## ✅ Terms and Conditions
<details>
 <summary> Participation Terms (click to expand)</summary>

If not otherwise defined in these Participation Terms, capitalized terms contained herein shall have the meaning given 
to them in the Ultralytics’s Terms of Use available at [https://ultralytics.com/terms-of-use](https://ultralytics.com/terms-of-use) (the **ToU**).

## General

**1.1. Acceptation of the Participation Terms.**

By signing up as a Participant, you agree to the following Participation Terms, 
which are incorporated by reference in the ToU.

**1.2. Challenge Rules.** 

In addition to these Participation Terms and the ToU, Challenge launched on the Platform shall 
be governed by a set of ad hoc Challenge Rules, which are maintained by Ultralytics, and published on the Github
repository https://github.com/ultralytics/yolov5 

By making any submission for a Challenge, you agree to be bound by the applicable Challenge Rules.

**1.3. Order of Precedence.** 

In the event of conflict between the ToU, these Participation Terms and the applicable Challenge 
Rules, the order of precedence shall be as follow:

(i) the Challenge Rules;
(ii) these Participation Terms;
(iii) the ToU.
 
## Independent Parties

**2.1. In relation to Participants.** 

As Participants, you participate to the Challenge strictly as independent third-party 
and not as an employee, agent or partner of Ultralytics. You are not authorized to assume or create 
any obligation or responsibility, express or implied, on behalf of, or in the name of, or to bind, Ultralytics in any manner. 
The following conditions therefore apply:

a) you are not obliged to accept or participate to the Challenge and Ultralytics does not guarantee that you will be able 
to participate to Challenge;
b) you are solely responsible for all taxes, unemployment insurance and social security taxes in connection with the Challenge, 
including in relation to any Rewards granted, and for maintaining adequate workers’ compensation insurance coverage, if 
and to the extent required by applicable law;

c) you bear all the costs and expenses incurred in connection with each Challenge to which you participate and shall have 
no claim to reimbursement of costs and/or expenses incurred in connection therewith;

d) you are solely responsible for providing your own computer equipment, internet connection, data storage and software 
in order to participate to a Challenge and to pay any cost incurred in connection with your participation to a Challenge; and

e) Ultralytics has no obligation to provide you with any training, technical or administrative support in connection with the Challenge.

**2.2. In relation to Organizer.**

You acknowledge and agree that:

a) Organizers are employees, agents or partners of Ultralytics, or independent third parties which request the assistance of 
Ultralytics to setup and run a Challenge;

b) Ultralytics is responsible for the payment of the rewards listed for its Challenge (Rewards) as set forth and subject to 
the applicable Challenge Rules, where the case may be with the assistance of Ultralytics acting on behalf of the Organizer only. 
Ultralytics makes no warranty and assumes no liability in relation to the payment of such Rewards. In particular, Ultralytics 
is not responsible for delays in payment outside of its reasonable control; and

c) any legal or judicial remedy that you seek to obtain for an Organizer’s actions or omissions or other third parties 
regarding a Challenge, including the payment of any Rewards, shall be limited to a claim against the particular Organizer 
or other responsible third parties, and you shall not make any claim or attempt to impose any liability on Ultralytics 
with respect to those actions or omissions.
 
## Challenges and Rewards

**3.1. Right to Participate.**

Any User having opened an Account as Participant will have the right to participate to the Challenges, 
subject to specific restrictions set out in the applicable Challenge Rules.

**3.2. Changes to the Challenge.**

Projects based on artificial intelligence and machine learning often need to evolve or be adapted, 
for instance due to technology or legal requirements or for any other reason. You acknowledge that the specifications of 
each Challenge may change during its term. Ultralytics reserves the right to adapt the Challenge Rules to the extent it 
deems necessary in which case you will be informed by any appropriate means (including through the Platform or via email). 
Participants taking part in a Challenge may be evaluated on the basis of their ability to adapt to changes of the Challenge.

**3.3. Rewards.**

Reward details vary for each Challenge. Rewards can take the form of monetary prizes in USD, or any other 
currency, of crypto-assets, as well as of non-monetary rewards.

**3.4. Submissions’ Evaluation and Winners.** 

Each submission is evaluated by the Organizer responsible for the Challenge on 
the basis of the criteria set out in the Challenge Rules. Winners are chosen solely by the Organizer of the Challenge 
Ultralytics may assist the Organizer in the evaluation process but has no decision-making authority.

**3.5. No Right to Reward.**

Participants have no claim to receive any Reward. The Ultralytics has the right to make 
the final decision at its entire discretion on the winner of a given Challenge. All results are final, without any right 
of recourse or objections.

**3.6. Rewards eligibility.**

To be eligible to receive a Reward, you must:
a) provide Ultralytics with accurate, complete and up-to-date information about you, including your mailing address and 
any other information that Ultralytics reasonably requests, to allow the Organizer, as the case may be with the assistance 
of Ultralytics, to legally send any Reward to you and file the appropriate tax documentation, if and to the extend required;

b) comply with any conditions and requirements set out in the applicable Challenge Rules; and

c) do any other action or sign any document as may be reasonably necessary to comply with the above.
 

## Intellectual Property Rights

**4.1. Lawfulness.**

Any submission of Submission Materials in relation to a Challenge shall be made in accordance with, and 
pursuant to, the ToU and the applicable Challenge Rules. In particular, you represent that the Submission Materials you 
submit will not infringe, misappropriate or violate a third party’s intellectual property rights, or rights of publicity 
or privacy, or result in the violation of any applicable law or regulation, including export control laws. You must 
clearly inform upon submitting your Submission Materials of any third party materials (e.g. which are licensed from a third 
party, or available under an open source license or another licensing terms) embedded in or otherwise used in connection 
with your Submission Materials and provide a description of such third party materials and licensing terms.

**4.2. Ownership and License.**

Unless otherwise provided for in the applicable Challenge Rules, Ultralytics does not claim 
any ownership rights in the Submission Materials you submit in connection with a Challenge. However, by making any 
Submission Materials available to Ultralytics through the Services, you hereby grant to Ultralytics, its assignees and 
sublicensees (including Organizers) a worldwide, free of charge, transferable license to use your Submission Materials 
strictly for the purpose of providing the Services (including running Challenges). Any intellectual property rights 
generated through the use by Ultralytics of the Submission Materials shall be the sole property of Ultralytics.

**4.3. Software Release.**

Except as otherwise provided for in the applicable Challenge Rules, you have the right to decide 
which licensing terms must apply to the software code (including the associated documentation) contained in your 
Submission Materials, by specifying upon your submission the licensing terms applicable thereto. If you do not specify 
otherwise upon your submission, or do not provide the applicable licensing terms, you hereby agree that the MIT open 
source license (https://opensource.org/licenses/MIT) will apply to all software code contained in your Submission Materials 
and authorize Ultralytics to affix the applicable copyright notice to your Submission Materials if such notice is missing. 
For the avoidance of doubt, the license granted pursuant to clause 4.2 above is independent from, and apply notwithstanding, 
the licensing terms pursuant to this clause 4.3.

**4.4. Challenge’s winners.**

In addition to the licenses pursuant to clauses 4.2 and 4.3 above, the applicable Challenge 
Rules may require that, in order to be eligible to receive a Reward, winners must accept to grant to the Challenge's 
Organizer additional rights to their Submission Materials (such as the right to commercially use such Submission Materials). 
 
## Miscellaneous

**5.1. Communications.**

All communications relating to a Challenge shall be made exclusively through the Platform or as 
otherwise specified in the applicable Challenge Rules. You are not allowed to directly communicate with an Organizer 
in relation to an ongoing Challenge to which you are taking part, except as otherwise specified in the applicable Challenge 
Rules.

**5.2. No Liability.**

You confirm that your participation to a Challenge does not put you in breach of any agreement you 
may be bound by, in particular but not limited to, any breach of any employment contract, consultancy contract or 
confidentiality contract with any third party.

**5.3. Applicable laws and Jurisdiction.**

As per the ToU.

</details>

## 🤗 Share 

Help us spread the word! Tweet about the competition and let your friends know. The more participants the better the results will be for the community!

<a href="https://twitter.com/intent/tweet?text=Make YOLOv5 🚀 even better and win up to $10,000.00 in cash prizes! Help us and spread the word about the first-ever Ultralytics YOLOv5 🚀 EXPORT Competition! &url=https://github.com/ultralytics/yolov5&hashtags=YOLOv5&original_referer=https://github.com&tw_p=tweetbutton" target="_blank"><img width="96" alt="Share on Twitter" src="https://user-images.githubusercontent.com/26833433/118511342-1b590a00-b732-11eb-821a-2af41ee70204.png"></a>
<a href="https://www.linkedin.com/shareArticle?mini=true&url=https://github.com/ultralytics/yolov5"><img width="96" alt="Share on LinkedIn" src="https://user-images.githubusercontent.com/26833433/118512908-7c351200-b733-11eb-9c20-cb4b921908d5.png"></a>
<a href="https://www.facebook.com/sharer/sharer.php?u=https://github.com/ultralytics/yolov5"><img width="96" alt="Share on FaceBook" src="https://user-images.githubusercontent.com/26833433/118512914-7ccda880-b733-11eb-8c80-e22f4315bc99.png"></a>

**Comment by benjamingriff** [comment]
Where are the official competition submission threads?  😃

**Reply by batrlatom** [reply]
You can submit in one of the discussions:

[Nvidia Jetson Nano](https://github.com/ultralytics/yolov5/discussions/3426)
[Google Edge TPU](https://github.com/ultralytics/yolov5/discussions/3428)
[Raspbery Pi](https://github.com/ultralytics/yolov5/discussions/3430)
[Intel/AMD CPU](https://github.com/ultralytics/yolov5/discussions/3432)
[Android](https://github.com/ultralytics/yolov5/discussions/3434)

Good luck!

**Reply by benjamingriff** [reply]
Thanks for setting these up. I'm hoping to make a submission to the Raspberry Pi category. Are we able to use external hardware in the tutorial to accelerate predictions e.g. Intel Movidius Neural Network Compute Stick?

Thanks 👍

**Reply by batrlatom** [reply]
Movidius Neural Network Compute Stick is a great piece of hardware. But users mostly want inference on bare Raspi, so we decided not to include this accelerator in the competition for now. You are welcome to make a submission for an ordinary Raspberry, I believe it would be very useful. But if your solution will work on normal Raspberry and with Compute stick (Myriad X) on top of that, it would receive extra scoring points!

**Comment by RioRocker97** [comment]
This look like a new journey for me . I have some experienced working around with your YOLOv5 hoping it could help me with the competition

**Reply by batrlatom** [reply]
We look forward to your contribution to the competition, Rio!

**Comment by henbucuoshanghai** [comment]
how to put the yolo5  on my onw xioami phone?

**Reply by glenn-jocher** [reply]
@henbucuoshanghai if you have a good YOLOv5 Android implementation please submit to our Android section at https://github.com/ultralytics/yolov5/discussions/3434. Thank you!

**Reply by batrlatom** [reply]
@henbucuoshanghai If you are new to the yolov5 and just want to try it on your phone, you can take a look on the official tutorial app from pytorch team at [example tutorial](https://pytorch.org/mobile/android/) and [example repo](https://github.com/pytorch/android-demo-app/tree/master/ObjectDetection)

**Comment by KIRNESH** [comment]
Can we participate in Teams? If yes, how to do that?

**Reply by batrlatom** [reply]
If you want to participate as a team, just choose a team leader who will post a submission for the whole team. In case the team will win, we can send a prize to the leader and he will share it with the rest.

**Comment by mshamash** [comment]
Any chance an iOS app for the biomedical field developed using a YOLOv5 model can be considered? Currently working on a project which I'd like to submit, though it is iOS only for now...

**Reply by batrlatom** [reply]
Hi, @mshamash . We already have an export to CoreML developed, so we decided not to include export to iOS in the competition.

**Reply by mshamash** [reply]
Makes sense, thanks for the clarification!

**Comment by AnhPC03** [comment]
```For Android deployments, an Android reference app should also be included```
So, I need to build an Android app? I don't really understand your above sentence. Could you clarify me? Thank you.

**Reply by batrlatom** [reply]
We require only source code for the app. If you build it and provide APK, it would be great

**Comment by rendinurcahyo22** [comment]
how to register to this competition? we just need to submit if we have finished?

**Reply by batrlatom** [reply]
Hi, just send a link of your submission to the category you want to participate and you are all set.

**Reply by rendinurcahyo22** [reply]
okay. Thanks for the response

**Comment by hovhanns** [comment]
Hi, I have a question about Jetson's submission. So the goal is to run the YOLOv5 on Jetson, right? I was able to run the model on Jetson dealing with various dependency problems. I mean is it enough to solve those problems and submit them? So running the script on any Jetson can install necessary libs and run the inference.
Thanks.

**Reply by batrlatom** [reply]
Hi, the goal is to having yolov5 running on jetson as fast and precise as possible. Making it running is just half of the work. The right direction is to make it running with tensorRT engine and I think that are almost all libraries you need already installed with jetpack 4.5.1 . But fell free to submit whatever you find appropriate nad working. All submission counts! Good luck

**Comment by jaskiratsingh2000** [comment]
@batrlatom @glenn-jocher I just have a key question that what are we trying to do with exportation of current Yolov5 into the exported version? What is the need to export for Raspberry Pi like devices?

What is the difference between current yolov5 and the exported version of yolov5?

Can anyone of you please let me know? I would highly appreciate that.

**Reply by antran89** [reply]
Hi @batrlatom, @glenn-jocher, I have additional questions. What is the procedure to test it on Raspberry device? Thanks.

**Reply by batrlatom** [reply]
Hi @antran89, the general procedure is described in the description in the scoring section. we will try to go through the procedure that each competitor should have described. Accuracy and speed are just some of the more criteria. It will also depend on the quality of the documentation and also on ease of use.

**Comment by jaskiratsingh2000** [comment]
HI, @batrlatom By when these export would be available for use with all the functionalities as there are currently in the present version?

**Reply by batrlatom** [reply]
the winners will be announced sometime in the second half of September. But now you can visit the raspber section of the category and test the results of the people who participate in the competition.

**Reply by jaskiratsingh2000** [reply]
Thanks, @batrlatom What I was looking forward to was the mAP value(Accuracy Performance) and none of these submissions have implemented that. SO I don't think so if there is not a priority on the Accuracy Performance then what is the use of export.

**Comment by karanjakhar** [comment]
@batrlatom any updates regarding results?

**Reply by batrlatom** [reply]
we will announce results soon!

**Reply by jaskiratsingh2000** [reply]
@karanjakhar BTW, it's been so long and after multiple pings, you haven't answered my question on your submission repo. Not sure why so but if you can really answer my question regarding mAP value that would show more of a open-source spirit. Thanks

**Comment by stefani-kovachevska** [comment]
Hi participants, 

Thank you all for your submissions, we have already reviewed them and are about to announce the winners! Please **send me an email** at [stefani@ultralytics.com](mailto:stefani@ultralytics.com) including your full name and GitHub username, so I can share more information with you. 

Stay tuned - the results are coming soon!

Stefani Kovachevska
_Project Manager at  Ultralytics_

**Reply by stefani-kovachevska** [reply]
We're pleased to announce the winners of the 2021 YOLOv5 Export Competition for each of the categories. 
Congratulations to @alxmamaev, @jveitchmichaelis and @lp6m! 🎉 

We also want to send a big thank you to everyone who participated in our Export Competition! We are fortunate to have numerous valuable members of our open-source community. It's the contributions from all of you that make our community great! 

Check out our latest blog post for more details: [https://ultralytics.com/article/export-competition-winners](https://ultralytics.com/article/export-competition-winners)

Stay amazing and keep creating! 🚀 
_Team Ultralytics_
